packages/camera/camera_platform_interface/lib/src/events/camera_event.dart

```diff
@@ -27,26 +25,84 @@ abstract class CameraEvent {
 
 /// An event fired when the resolution preset of the camera has changed.
 class ResolutionChangedEvent extends CameraEvent {
-  /// The capture size in pixels.
-  final Size captureSize;
+  /// The capture width in pixels.
+  final int captureWidth;
+
+  /// The capture height in pixels.
+  final int captureHeight;
 
-  /// The size of the preview in pixels.
-  final Size previewSize;
+  /// The width of the preview in pixels.
+  final int previewWidth;
+
+  /// The height of the preview in pixels.
+  final int previewHeight;
 
   /// Build a ResolutionChanged event triggered from the camera represented by
   /// `cameraId`.
   ///
   /// The `captureSize` represents the size of the resulting image in pixels.
   /// The `previewSize` represents the size of the generated preview in pixels.
-  ResolutionChangedEvent(int cameraId, this.captureSize, this.previewSize)
-      : super(cameraId);
+  ResolutionChangedEvent(
+    int cameraId,
+    this.captureWidth,
+    this.captureHeight,
+    this.previewWidth,
+    this.previewHeight,
+  ) : super(cameraId);
+
+  ResolutionChangedEvent.fromJson(Map<String, dynamic> json)
+      : captureWidth = json['captureWidth'],
+        captureHeight = json['captureHeight'],
+        previewWidth = json['previewWidth'],
+        previewHeight = json['previewHeight'],
+        super(json['cameraId']);
+
+  Map<String, dynamic> toJson() => {
+        'cameraId': cameraId,
+        'captureWidth': captureWidth,
+        'captureHeight': captureHeight,
+        'previewWidth': previewWidth,
+        'previewHeight': previewHeight,
+      };
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is ResolutionChangedEvent &&
+          runtimeType == other.runtimeType &&
+          captureWidth == other.captureWidth &&
+          captureHeight == other.captureHeight &&
+          previewWidth == other.previewWidth &&
+          previewHeight == other.previewHeight;
```

Comment:
We should also include the `cameraId` from the base class in this comparison. I am not sure how to best handle this, maybe it is enough to override the `==` operator and `hashCode` property in the base class (being `CameraEvent`). 

Should be good to quickly research this.